packages/api/src/apps/datastore_api/app.ts

```diff
@@ -297,10 +264,13 @@ export default async (env: ProcessEnv) => {
     )
       .catch(console.error)
       .finally(end);
+    appState.loopsLock = false;
   }, detectContractsUpdateRateS * 1000);
 
   // main update loop for all state, executes immediately and waits for updateRateS
   utils.loop(async () => {
+    if (appState.loopsLock) return;
```

Comment:
tbh im not a huge fan of this solution. it only really matters that we detect contracts before we check the state for the first time.  maybe we just initialize the second loop after the first runs or something. or maybe we run the functions in the loops in order and then init the loops, similar to how it was before

Reply:
I switched back to the older implementation with running the functions before the loops. Just re-organised the code a little bit. Maybe you can double check the functions execution order